### DISCUSSION TITLE
Why does Prometheus consume so much memory?

### DISCUSSION THREAD

**Discussion by ijihang** [prometheus/prometheus#9903] [Q&A] [answered]
I'm using Prometheus 2.14.0 for monitoring a large environment of nodes，I've noticed that the Prometheus process consumes more and more memory，How should I limit the memory of Prometheus?
Thanks!

**Comment by codesome** [comment, marked as answer]
Please use the latest version v2.31.1. You will immediately see a huge reduction in memory usage.

And the memory usage grows with increase in number of active time series.
